src/Common/dataAccess/deleteDocument.ts

```diff
@@ -24,3 +25,58 @@ export const deleteDocument = async (collection: CollectionBase, documentId: Doc
     clearMessage();
   }
 };
+
+/**
+ * Bulk delete documents
+ * @param collection
+ * @param documentId
+ * @returns array of ids that were successfully deleted
+ */
+export const deleteDocuments = async (collection: CollectionBase, documentIds: DocumentId[]): Promise<DocumentId[]> => {
+  const nbDocuments = documentIds.length;
+  const clearMessage = logConsoleProgress(`Deleting ${documentIds.length} ${getEntityName(true)}`);
+  try {
+    const v2Container = await client().database(collection.databaseId).container(collection.id());
+
+    // Bulk can only delete 100 documents at a time
+    const BULK_DELETE_LIMIT = 100;
+    const promiseArray = [];
+
+    while (documentIds.length > 0) {
+      const documentIdsChunk = documentIds.splice(0, BULK_DELETE_LIMIT);
+      const operations: OperationInput[] = documentIdsChunk.map((documentId) => ({
+        id: documentId.id(),
+        // bulk delete: if not partition key is specified, do not pass empty array, but undefined
+        partitionKey:
+          documentId.partitionKeyValue &&
+          Array.isArray(documentId.partitionKeyValue) &&
+          documentId.partitionKeyValue.length === 0
+            ? undefined
+            : documentId.partitionKeyValue,
+        operationType: BulkOperationType.Delete,
+      }));
+
+      const promise = v2Container.items.bulk(operations).then((bulkResult) => {
+        const result: DocumentId[] = [];
+        documentIdsChunk.forEach((documentId, index) => {
+          if (bulkResult[index].statusCode === 204) {
+            result.push(documentId);
+          }
+        });
+        return result;
```

Comment:
non-blocking style option: This could be a filter.

```suggestion
        return documentIdsChunk.filter((_, index) => bulkResult[index].statusCode === 204);
```